spyder/api/plugins/new_api.py

```diff
@@ -491,76 +781,110 @@ def get_dockable_plugins(self):
+            case in tests).
+
+        Raises
+        ------
+        SpyderAPIError
+            If ``section`` is not passed and :attr:`CONF_SECTION` is not
+            defined.
+        configparser.NoOptionError
+            If the ``section`` does not exist in Spyder's configuration.
+        """
+        if self._conf is None:
+            return None
+
+        section = self.CONF_SECTION if section is None else section
+        if section is None:
+            raise SpyderAPIError(
+                "A spyder plugin must define a `CONF_SECTION` class "
+                "attribute!"
+            )
+        return self._conf.get(section, option, default, secure=secure)
 
     @Slot(str, object)
     @Slot(str, object, str)
     def set_conf(
         self,
-        option,
-        value,
-        section=None,
-        recursive_notification=True,
-        secure=False,
-    ):
+        option: spyder.config.types.ConfigurationKey,
+        value: BasicTypes,
+        section: str | None = None,
+        recursive_notification: bool = True,
+        secure: bool = False,
+    ) -> None:
         """
-        Set an option in Spyder configuration system.
+        Set an option's value in the Spyder configuration system.
 
         Parameters
         ----------
-        option: str
-            Name of the option (e.g. 'case_sensitive')
-        value: bool, int, str, tuple, list, dict
-            Value to save in the configuration system, passed as a
-            Python object.
-        section: str
-            Section in the configuration system, e.g. `shortcuts`.
-        recursive_notification: bool
-            If True, all objects that observe all changes on the
-            configuration section and objects that observe partial tuple paths
-            are notified. For example if the option `opt` of section `sec`
-            changes, then the observers for section `sec` are notified.
-            Likewise, if the option `(a, b, c)` changes, then observers for
-            `(a, b, c)`, `(a, b)` and a are notified as well.
-        secure: bool
-            If True, the option will be saved securely using the `keyring`
-            Python package.
+        option: spyder.config.types.ConfigurationKey
+            Name/tuple path of the configuration option to set.
+        value: spyder.api.config.mixins.BasicTypes
+            Value to set for the given configuration option.
+        section: str | None, optional
+            Name of the configuration section to use, e.g. ``"shortcuts"``.
+            If ``None``, the default, then the value of :attr:`CONF_SECTION`
+            is used.
+        recursive_notification: bool, optional
+            If ``True``, all objects that observe all changes on the
+            configuration ``section`` as well as objects that observe
+            partial tuple paths are notified. For example, if the
+            ``option`` ``"opt"`` of ``section`` ``"sec"`` changes, then
```

Comment:
```suggestion
            ``option`` `opt" of ``section`` "sec" changes, then
```

I find this is styling confusing in the rendered docs, so I think it's better to remove it:

<img width="674" height="83" alt="image" src="https://github.com/user-attachments/assets/435b1e1b-bfef-47a7-968d-0946810b1afb" />